pkg/processor/trigger/kafka/types.go

```diff
@@ -157,6 +157,18 @@ func NewConfiguration(id string,
 
 	newConfiguration.WorkerAllocationMode = partitionworker.AllocationMode(workerAllocationModeValue)
 
+	// default explicit ack mode to 'disable'
+	if triggerConfiguration.ExplicitAckMode == "" {
+		newConfiguration.ExplicitAckMode = functionconfig.ExplicitAckModeDisable
+	}
+
+	// explicit ack is only allowed for Static Allocation mode
+	if newConfiguration.WorkerAllocationMode == partitionworker.AllocationModePool &&
+		(triggerConfiguration.ExplicitAckMode == functionconfig.ExplicitAckModeEnable ||
+			triggerConfiguration.ExplicitAckMode == functionconfig.ExplicitAckModeExplicitOnly) {
+		return nil, errors.New("Explicit ack mode is not allowed when using worker pool allocation mode")
+	}
```

Comment:
legit place to put it, but id also add this kind of validation to dashboard's abstract to get feedback before deployment even starts. that way, user saves couple of minutes before seeing this error.